src/app/components/audio-recordings/pages/download/download.component.ts

```diff
@@ -0,0 +1,389 @@
+import { Component, OnInit } from "@angular/core";
+import { ActivatedRoute } from "@angular/router";
+import { AudioRecordingsService } from "@baw-api/audio-recording/audio-recordings.service";
+import { Filters } from "@baw-api/baw-api.service";
+import { projectResolvers } from "@baw-api/project/projects.service";
+import { regionResolvers } from "@baw-api/region/regions.service";
+import {
+  hasResolvedSuccessfully,
+  ResolvedModelList,
+  retrieveResolvers,
+} from "@baw-api/resolver-common";
+import { siteResolvers } from "@baw-api/site/sites.service";
+import { contactUsMenuItem } from "@components/about/about.menus";
+import {
+  audioRecordingMenuItems,
+  audioRecordingsCategory,
+} from "@components/audio-recordings/audio-recording.menus";
+import { myAccountMenuItem } from "@components/profile/profile.menus";
+import { PageComponent } from "@helpers/page/pageComponent";
+import { IPageInfo, PageInfo } from "@helpers/page/pageInfo";
+import { withUnsubscribe } from "@helpers/unsubscribe/unsubscribe";
+import { Id } from "@interfaces/apiInterfaces";
+import { AudioRecording } from "@models/AudioRecording";
+import { Project } from "@models/Project";
+import { Region } from "@models/Region";
+import { Site } from "@models/Site";
+import { FormlyFieldConfig } from "@ngx-formly/core";
+import { ConfigService } from "@services/config/config.service";
+import { DateTime } from "luxon";
+
+const projectKey = "project";
+const regionKey = "region";
+const siteKey = "site";
+
+interface Model {
+  projectId?: Id;
+  regionId?: Id;
+  siteId?: Id;
+  recordingStartedBefore?: string;
+  recordingFinishedBefore?: string;
+}
+
+@Component({
+  selector: "baw-download",
+  template: `
+    <ng-container *ngIf="!failure">
```

Comment:
If there is a failure, the error component will be shown, and this component should show nothing

Reply:
this is not logic that belongs in this component. Whether or not to render a component should be determined by the container of the component

I suspect this is a larger pattern, make an issue to deal with it. Otherwise fix here.

Reply:
Moved to here: https://github.com/QutEcoacoustics/workbench-client/pull/1766#issuecomment-1020798283